gapic/samplegen/samplegen.py

```diff
@@ -86,48 +92,147 @@ class InvalidRequestSetup(SampleError):
     pass
 
 
-class DuplicateInputParameter(SampleError):
-    pass
+class Validator:
+    """
+    Class that validates samples.
 
+    Contains methods that validate different segments of a sample and maintains
+    state that's relevant to validation across different segments.
+    """
 
-class ResponseValidator:
     COLL_KWORD = "collection"
     VAR_KWORD = "variable"
     MAP_KWORD = "map"
     KEY_KWORD = "key"
     VAL_KWORD = "value"
     BODY_KWORD = "body"
 
-    def __init__(self, response):
+    def __init__(self):
+        """
+        Args:
```

Comment:
`__init__` only has `self` as an arg?

Reply:
Yes: it's sole purpose is to set up the `var_defs_` instance attr.
Your comment above about redefining `input_param`s made me realize that this makes a good case for an object, because state needs to persist across function calls, and doing that through input param/return value and higher level code stitching felt gross and inelegant.

Reply:
Sorry, I should have been more explicit and less rhetorical. I left the comment on this line because I was wondering why the docstring has an `Args` section given that `__init__` only has `self` as an arg.

Reply:
Argh, sloppy/paste on my part.